lib/websocket-server.js

```diff
@@ -330,17 +330,22 @@ function socketOnError () {
  * @param {net.Socket} socket The socket of the upgrade request
  * @param {Number} code The HTTP response status code
  * @param {String} [message] The HTTP response body
+ * @param {Object} [headers] Additional HTTP response headers
  * @private
  */
-function abortHandshake (socket, code, message) {
+function abortHandshake (socket, code, message, headers) {
   if (socket.writable) {
     message = message || http.STATUS_CODES[code];
+    headers = Object.assign({}, {
```

Comment:
The first `{}` is not needed. I also wonder if it's better to move the defaults after the custom headers to prevent them from being overridden.